scripts/semantic-release/tests/intents.spec.ts

```diff
@@ -0,0 +1,152 @@
+import { describe, it, expect } from '@jest/globals'
+import { PublicKey } from '@solana/web3.js'
+import {
+  encodeReward,
+  decodeReward,
+  VmType,
+  type EvmRewardType,
+  type SvmRewardType,
+} from '../assets/utils/intent'
+
+describe('Intent Utils - Reward Encoding/Decoding', () => {
+  describe('EVM Reward Encoding/Decoding', () => {
+    it('should encode and decode EVM reward correctly', () => {
+      const evmReward: EvmRewardType = {
+        vm: VmType.EVM,
+        creator: '0x1234567890123456789012345678901234567890' as `0x${string}`,
+        prover: '0x9876543210987654321098765432109876543210' as `0x${string}`,
+        deadline: 1735689600n,
+        nativeAmount: 1000000000000000000n,
+        tokens: [
+          {
+            token: '0xA0b86a33E6441e7c34A88A39bEAD93bB31d4fc6F' as `0x${string}`,
+            amount: 500000000000000000n,
+          },
+          {
+            token: '0xB1c86a33E6441e7c34A88A39bEAD93bB31d4fc6F' as `0x${string}`,
+            amount: 250000000000000000n,
+          },
+        ],
+      }
+
+      const encoded = encodeReward(evmReward)
```

Comment:
Should add `expect(encoded).toMatchSnapshot();` and all the other tests in this `describe`. The issue right now is that, if we update our encoding/decoding dependencies, chances are these tests are still gonna pass but changes to the `encoded` may be unexpected. We need the tests to catch that kinda change and decide if the external libs are buggy and if the contracts match the versioning.